bash_scripts/run_reencode_and_frame_extraction_array.sh

```diff
@@ -0,0 +1,166 @@
+# assumes repo located at '/ceph/scratch/sminano'
+SCRATCH_PERSONAL_DIR=/ceph/scratch/sminano
+SCRIPT_DIR=$SCRATCH_PERSONAL_DIR/crabs-exploration/crabs/bboxes_labelling
+
+# -------------------
+# Run python script
+# -------------------
+for i in {1..${SLURM_ARRAY_TASK_COUNT}}
+do
+    # Input video
+    SAMPLE=${INPUT_DATA_LIST[${SLURM_ARRAY_TASK_ID}]}
+    echo "Input video: $SAMPLE"
+    echo "--------"
+
+    # Reencode video if required
+    # following SLEAP's recommendations
+    # https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format
+    if [ "$flag_reencode_input_videos" = true ] ; then
+        echo "Rencoding ...."
+
+        # path to reencoded video
+        filename_no_ext="$(basename "$SAMPLE" | sed 's/\(.*\)\..*/\1/')" # filename without extension
+        REENCODED_VIDEO_PATH="$REENCODED_VIDEOS_SUBDIR/$filename_no_ext"_RE.$reencoded_extension
+
+        ffmpeg -version  # print version to logs
+        ffmpeg -y -i "$SAMPLE" \
+        -c:v libx264 \
+        -pix_fmt yuv420p \
+        -preset superfast \
+        -crf 15 \
+        $REENCODED_VIDEO_PATH
+
+
+        echo "Reencoded video: $REENCODED_VIDEO_PATH"
+        echo "--------"
+        FRAME_EXTRACTION_INPUT_VIDEO=$REENCODED_VIDEO_PATH
+    else
+        echo "Skipping video reencoding..."
+        echo "--------"
+        FRAME_EXTRACTION_INPUT_VIDEO=$SAMPLE
+    fi
+
+    # Get extension of input video
+    video_filename=$(basename -- "$FRAME_EXTRACTION_INPUT_VIDEO")
+    VIDEO_EXT="${video_filename##*.}"
+
+    # Run frame extraction algorithm on video
+    python $SCRIPT_DIR/extract_frames_to_label_w_sleap.py \
+    $FRAME_EXTRACTION_INPUT_VIDEO \
+    --output_path $OUTPUT_DIR \
+    --output_subdir $OUTPUT_SUBDIR \
+    --video_extensions $VIDEO_EXT \
+    --initial_samples $PARAM_INI_SAMPLES \
+    --scale $PARAM_SCALE \
+    --n_components $PARAM_N_COMPONENTS \
+    --n_clusters $PARAM_N_CLUSTERS \
+    --per_cluster $PARAM_PER_CLUSTER \
+    --compute_features_per_video
+
+    if [ "$?" -ne 0 ]; then
+        echo "Frame extraction failed! Please check .err log"
+    else
+        echo "Frames extracted from video: $FRAME_EXTRACTION_INPUT_VIDEO"
+    fi
+    echo "--------"
+
+    # Reencoded videos log
+    # copy .err file to go with reencoded video too if required
+    # filename: {reencoded video name}.{slurm_array}.{slurm_job_id}
+    # TODO: make a nicer log
+    if [ "$flag_reencode_input_videos" = true ] ; then
+        for ext in err out
+        do
+            cp slurm_array.$SLURMD_NODENAME.$SLURM_ARRAY_JOB_ID-$SLURM_ARRAY_TASK_ID.$ext \
+            /$REENCODED_VIDEOS_SUBDIR/"$filename_no_ext"_RE.slurm_array.$SLURM_ARRAY_JOB_ID-$SLURM_ARRAY_TASK_ID.$ext
+        done
+    fi
+
+    # Frame extraction logs
+    # Move logs for this job to subdir with extracted frames
+    for ext in err out
+    do
+        mv slurm_array.$SLURMD_NODENAME.$SLURM_ARRAY_JOB_ID-$SLURM_ARRAY_TASK_ID.$ext /$LOG_DIR
+    done
```

Comment:
```suggestion
    mv slurm_array.$SLURMD_NODENAME.$SLURM_ARRAY_JOB_ID-$SLURM_ARRAY_TASK_ID.{err,out} /$LOG_DIR
```
.. no? (I think you need the `ext` variable in the loop above though.)

Reply:
p.s. please test. It's been ages since I wrote bash scripts for a cluster.

Reply:
yeah it seems `cp` doesn't admit parameter expansion in the same way as `mv`
see [here](https://stackoverflow.com/a/5580894)